### ISSUE TITLE
Figure stops auto-rescaling or crashes when resizing GUI window after `save` with `GLMakie`

### ISSUE THREAD

**Issue by haakon-e** [MakieOrg/Makie.jl#2425]
### TLDR;

It appears figure rescaling works poorly with figure `save`ing.
1. If I create a figure from REPL and save it, auto-rescaling of the figure when resizing the GUI window stops working.
2. If I put the same code in a script and `include` the script, attempting to resize the GUI window throws an error.

### In an interactive julia session:
```julia
julia> using GLMakie

julia> fig = Figure()

julia> ax = Axis(fig[1,1])
```
The figure automatically rescales when I resize the GUI window, as expected:
![ezgif com-gif-maker](https://user-images.githubusercontent.com/45243236/202307036-23284aca-234d-4c0d-92f8-f1a5d9629c11.gif)
However, If I now save the figure:
```julia
julia> save("fig.png", fig)
```
then the auto-rescaling on GUI resizing stops working:
![ezgif com-gif-maker-2](https://user-images.githubusercontent.com/45243236/202307702-4b6a7fdc-498b-4134-98c5-9965ceb01f22.gif)
If I re-generate the figure by
```julia
julia> fig
```
auto-rescaling works as expected again.

### If the same code is `include`-ed:
Suppose I have a file, `example.jl`:
```julia
# contents of example.jl

using GLMakie
fig = Figure()
ax = Axis(fig[1,1])
save("fig.png", fig)
fig
```
then in terminal, I type:
```julia
julia> include("example.jl")
```
If I try to resize the GUI window of the generated figure, I get this error:

<details><summary>Binding freed GLBuffer{Point{2, Float32}}</summary>

```julia
julia> Binding freed GLBuffer{Point{2, Float32}}
Stacktrace:
   [1] error(s::String)
     @ Base ./error.jl:35
   [2] bind(buffer::GLMakie.GLAbstraction.GLBuffer{Point{2, Float32}})
     @ GLMakie.GLAbstraction ~/.julia/packages/GLMakie/x68HK/src/GLAbstraction/GLBuffer.jl:24
   [3] gpu_data
     @ ~/.julia/packages/GLMakie/x68HK/src/GLAbstraction/GLBuffer.jl:101 [inlined]
   [4] gpu_resize!(buffer::GLMakie.GLAbstraction.GLBuffer{Point{2, Float32}}, newdims::Tuple{Int64})
     @ GLMakie.GLAbstraction ~/.julia/packages/GLMakie/x68HK/src/GLAbstraction/GLBuffer.jl:114
   [5] resize!
     @ ~/.julia/packages/GLMakie/x68HK/src/GLAbstraction/AbstractGPUArray.jl:134 [inlined]
   [6] resize!
     @ ~/.julia/packages/GLMakie/x68HK/src/GLAbstraction/AbstractGPUArray.jl:131 [inlined]
   [7] update!
     @ ~/.julia/packages/GLMakie/x68HK/src/GLAbstraction/AbstractGPUArray.jl:63 [inlined]
   [8] (::GLMakie.GLAbstraction.var"#10#11"{GLMakie.GLAbstraction.GLBuffer{Point{2, Float32}}})(x::Vector{Point{2, Float32}})
     @ GLMakie.GLAbstraction ~/.julia/packages/GLMakie/x68HK/src/GLAbstraction/AbstractGPUArray.jl:196
   [9] #invokelatest#2
     @ ./essentials.jl:729 [inlined]
  [10] invokelatest
     @ ./essentials.jl:726 [inlined]
  [11] notify
     @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
  [12] setindex!(observable::Observable, val::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
  [13] (::Observables.MapCallback)(value::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:431
--- the last 5 lines are repeated 1 more time ---
  [19] #invokelatest#2
     @ ./essentials.jl:729 [inlined]
  [20] invokelatest
     @ ./essentials.jl:726 [inlined]
  [21] notify
     @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
  [22] setindex!(observable::Observable, val::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
  [23] (::Makie.var"#175#177"{Attributes, Observable{Tuple{Vector{Point{2, Float32}}}}, DataType})(kwargs::Tuple{}, args::Vector{Point{2, Float32}})
     @ Makie ~/.julia/packages/Makie/iY5BJ/src/interfaces.jl:333
  [24] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
     @ Base ./essentials.jl:729
  [25] invokelatest(::Any, ::Any, ::Vararg{Any})
     @ Base ./essentials.jl:726
  [26] (::Observables.OnAny)(value::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:415
  [27] #invokelatest#2
     @ ./essentials.jl:729 [inlined]
  [28] invokelatest
     @ ./essentials.jl:726 [inlined]
  [29] notify
     @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
  [30] setindex!(observable::Observable, val::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
  [31] (::Makie.var"#1859#1863"{Scene, Observable{Vector{Point{2, Float32}}}, Base.RefValue{Vector{Int64}}})(segs::Vector{Point{2, Float32}}, pos::Vector{Point{2, Float32}}, #unused#::StaticArraysCore.SMatrix{4, 4, Float32, 16}, #unused#::GeometryBasics.HyperRectangle{2, Int64}, transf::Function)
     @ Makie ~/.julia/packages/Makie/iY5BJ/src/basic_recipes/text.jl:59
  [32] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
     @ Base ./essentials.jl:729
  [33] invokelatest(::Any, ::Any, ::Vararg{Any})
     @ Base ./essentials.jl:726
  [34] (::Observables.OnAny)(value::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:415
  [35] #invokelatest#2
     @ ./essentials.jl:729 [inlined]
  [36] invokelatest
     @ ./essentials.jl:726 [inlined]
  [37] notify
     @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
  [38] setindex!(observable::Observable, val::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
  [39] (::Observables.MapCallback)(value::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:431
  [40] #invokelatest#2
     @ ./essentials.jl:729 [inlined]
  [41] invokelatest
     @ ./essentials.jl:726 [inlined]
  [42] notify
     @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
  [43] setindex!(observable::Observable, val::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
  [44] (::Makie.var"#175#177"{Attributes, Observable{Tuple{Vector{Point{2, Float32}}}}, DataType})(kwargs::Tuple{}, args::Point{2, Float32})
     @ Makie ~/.julia/packages/Makie/iY5BJ/src/interfaces.jl:333
  [45] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
     @ Base ./essentials.jl:729
  [46] invokelatest(::Any, ::Any, ::Vararg{Any})
     @ Base ./essentials.jl:726
  [47] (::Observables.OnAny)(value::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:415
  [48] #invokelatest#2
     @ ./essentials.jl:729 [inlined]
  [49] invokelatest
     @ ./essentials.jl:726 [inlined]
  [50] notify
     @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
  [51] setindex!(observable::Observable, val::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
  [52] (::Observables.MapCallback)(value::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:431
--- the last 5 lines are repeated 1 more time ---
  [58] #invokelatest#2
     @ ./essentials.jl:729 [inlined]
  [59] invokelatest
     @ ./essentials.jl:726 [inlined]
  [60] notify
     @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
  [61] setindex!(observable::Observable, val::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
  [62] (::Observables.SetindexCallback)(x::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:111
  [63] #invokelatest#2
     @ ./essentials.jl:729 [inlined]
  [64] invokelatest
     @ ./essentials.jl:726 [inlined]
  [65] notify
     @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
  [66] setindex!(observable::Observable, val::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
  [67] (::Observables.MapCallback)(value::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:431
--- the last 5 lines are repeated 2 more times ---
  [78] #invokelatest#2
     @ ./essentials.jl:729 [inlined]
  [79] invokelatest
     @ ./essentials.jl:726 [inlined]
  [80] notify
     @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
  [81] setindex!(observable::Observable, val::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
  [82] update_computedbbox!(computedbbox::Observable{GeometryBasics.HyperRectangle{2, Float32}}, suggestedbbox::GeometryBasics.HyperRectangle{2, Float32}, alignment::Tuple{Float32, Float32}, reporteddimensions::GridLayoutBase.Dimensions, alignmode::Inside, protrusions::GridLayoutBase.RectSides{Float32}, sizeattrs::Observable{Tuple{Union{Nothing, Float32, Auto, Fixed, Relative}, Union{Nothing, Float32, Auto, Fixed, Relative}}}, autosizeobservable::Observable{Tuple{Union{Nothing, Float32}, Union{Nothing, Float32}}})
     @ GridLayoutBase ~/.julia/packages/GridLayoutBase/lYdxT/src/layoutobservables.jl:356
  [83] (::GridLayoutBase.var"#106#108"{Observable{GridLayoutBase.Dimensions}, Observable{Tuple{Union{Nothing, Float32, Auto, Fixed, Relative}, Union{Nothing, Float32, Auto, Fixed, Relative}}}, Observable{Tuple{Union{Nothing, Float32}, Union{Nothing, Float32}}}, Observable{Any}, Observable{GridLayoutBase.RectSides{Float32}}, Observable{GeometryBasics.HyperRectangle{2, Float32}}})(sbbox::GeometryBasics.HyperRectangle{2, Float32}, ali::Tuple{Float32, Float32})
     @ GridLayoutBase ~/.julia/packages/GridLayoutBase/lYdxT/src/layoutobservables.jl:218
  [84] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
     @ Base ./essentials.jl:729
  [85] invokelatest(::Any, ::Any, ::Vararg{Any})
     @ Base ./essentials.jl:726
  [86] (::Observables.OnAny)(value::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:415
  [87] #invokelatest#2
     @ ./essentials.jl:729 [inlined]
  [88] invokelatest
     @ ./essentials.jl:726 [inlined]
  [89] notify
     @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
  [90] setindex!(observable::Observable, val::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
  [91] align_to_bbox!(gl::GridLayout, suggestedbbox::GeometryBasics.HyperRectangle{2, Float32})
     @ GridLayoutBase ~/.julia/packages/GridLayoutBase/lYdxT/src/gridlayout.jl:1084
  [92] (::GridLayoutBase.var"#11#12"{GridLayout})(bbox::GeometryBasics.HyperRectangle{2, Float32})
     @ GridLayoutBase ~/.julia/packages/GridLayoutBase/lYdxT/src/gridlayout.jl:157
  [93] #invokelatest#2
     @ ./essentials.jl:729 [inlined]
  [94] invokelatest
     @ ./essentials.jl:726 [inlined]
  [95] notify
     @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
  [96] setindex!(observable::Observable, val::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
  [97] update_computedbbox!(computedbbox::Observable{GeometryBasics.HyperRectangle{2, Float32}}, suggestedbbox::GeometryBasics.HyperRectangle{2, Float32}, alignment::Tuple{Float32, Float32}, reporteddimensions::GridLayoutBase.Dimensions, alignmode::Inside, protrusions::GridLayoutBase.RectSides{Float32}, sizeattrs::Observable{Tuple{Union{Nothing, Float32, Auto, Fixed, Relative}, Union{Nothing, Float32, Auto, Fixed, Relative}}}, autosizeobservable::Observable{Tuple{Union{Nothing, Float32}, Union{Nothing, Float32}}})
     @ GridLayoutBase ~/.julia/packages/GridLayoutBase/lYdxT/src/layoutobservables.jl:356
  [98] (::GridLayoutBase.var"#106#108"{Observable{GridLayoutBase.Dimensions}, Observable{Tuple{Union{Nothing, Float32, Auto, Fixed, Relative}, Union{Nothing, Float32, Auto, Fixed, Relative}}}, Observable{Tuple{Union{Nothing, Float32}, Union{Nothing, Float32}}}, Observable{Union{Inside, Outside, Mixed}}, Observable{GridLayoutBase.RectSides{Float32}}, Observable{GeometryBasics.HyperRectangle{2, Float32}}})(sbbox::GeometryBasics.HyperRectangle{2, Float32}, ali::Tuple{Float32, Float32})
     @ GridLayoutBase ~/.julia/packages/GridLayoutBase/lYdxT/src/layoutobservables.jl:218
  [99] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
     @ Base ./essentials.jl:729
 [100] invokelatest(::Any, ::Any, ::Vararg{Any})
     @ Base ./essentials.jl:726
 [101] (::Observables.OnAny)(value::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:415
 [102] #invokelatest#2
     @ ./essentials.jl:729 [inlined]
 [103] invokelatest
     @ ./essentials.jl:726 [inlined]
 [104] notify
     @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
 [105] setindex!(observable::Observable, val::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
 [106] (::Observables.MapCallback)(value::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:431
 [107] #invokelatest#2
     @ ./essentials.jl:729 [inlined]
 [108] invokelatest
     @ ./essentials.jl:726 [inlined]
 [109] notify
     @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
 [110] setindex!(observable::Observable, val::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
 [111] (::Makie.var"#105#108")(w_area::GeometryBasics.HyperRectangle{2, Int64})
     @ Makie ~/.julia/packages/Makie/iY5BJ/src/scenes.jl:179
 [112] #invokelatest#2
     @ ./essentials.jl:729 [inlined]
 [113] invokelatest
     @ ./essentials.jl:726 [inlined]
 [114] notify
     @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
 [115] setindex!(observable::Observable, val::Any)
     @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
 [116] (::GLMakie.WindowAreaUpdater)(#unused#::Nothing)
     @ GLMakie ~/.julia/packages/GLMakie/x68HK/src/events.jl:64
 [117] #invokelatest#2
     @ ./essentials.jl:729 [inlined]
 [118] invokelatest
     @ ./essentials.jl:726 [inlined]
 [119] notify
     @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
 [120] setindex!
     @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86 [inlined]
 [121] refreshwindowcb(window::GLFW.Window, screen::GLMakie.Screen{GLFW.Window})
     @ GLMakie ~/.julia/packages/GLMakie/x68HK/src/screen.jl:623
 [122] (::GLMakie.var"#41#42"{GLMakie.Screen{GLFW.Window}})(window::GLFW.Window)
     @ GLMakie ~/.julia/packages/GLMakie/x68HK/src/screen.jl:268
julia> ^C
```

</details>

and the GUI window hangs until I completely close (`^D`) that julia session.

--------------

<details><summary>System info</summary>

```julia
(test) pkg> st
Status `~/test/Project.toml`
  [e9467ef8] GLMakie v0.7.2

julia> versioninfo()
Julia Version 1.8.2
Commit 36034abf260 (2022-09-29 15:21 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin21.3.0)
  CPU: 8 × Apple M1
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, apple-m1)
  Threads: 1 on 4 virtual cores
```

</details>

**Comment by SimonDanisch** [comment]
Without looking at the details, can you try master? I fixed quite a few of these...

**Comment by haakon-e** [comment]
Both of these seem to work on master, thanks!

**Comment by haakon-e** [comment]
Ah, I see this was a duplicate of #2380, I didn't catch that looking through past issues!
